planner/funcdep/fd_graph.go

```diff
@@ -39,6 +39,11 @@ type fdEdge struct {
 	conditionNC *FastIntSet
 }
 
+type conditionNotNull struct {
```

Comment:
Why we put these two into a struct?

Reply:
like ncEdge did.
```
type fdEdge struct {
	from FastIntSet
	to   FastIntSet
        .....
	conditionNC *FastIntSet
}
```
the edge with a condition (basically a column set waiting for one of them being null-rejected)
so as not-null columns
eg: the originally not-null column on the inner side, will be stored as conditionNotNull (one is original not-null, one is kind of null-constraints) waiting for one of their condition columns to be null-rejected, then this kind of not-null can be used again (make not null)